### PR TITLE
London Class 7 - Somayeh Karimi - JavaScript-Core1 - Week 3

### PR DESCRIPTION
**Pull request by DibaKarimi**
# Your Details

- Your Name:Somayeh Karimi
- Your City: London
- Your Slack Name: Somayeh Kr

# Homework Details

- Module: JavaScript - Core 1 
- Week: 3

### CODE REVIEW

File: 1-exercises/A-array-find/exercise.js [hunk 1]
```diff
@@ -16,9 +16,12 @@ var names = [
   "Karim",
   "Ahmed",
 ];
-
-var longNameThatStartsWithA = findLongNameThatStartsWithA(names);
-
+function isLongName(name) {
+  return name[0] === "A";
```
Comment (akbasmustafa) [inline 1-exercises/A-array-find/exercise.js:20]: Great work, I think you need to check length of name AND 'A' check

**Review by akbasmustafa — CHANGES_REQUESTED** [review]

File: 1-exercises/J-string-substring/exercise2.js [hunk 2]
```diff
@@ -14,11 +14,11 @@ let names = [
...
-names[0] = names[0].substring();
-names[1] = names[1].substring();
-names[2] = names[2].substring();
-names[3] = names[3].substring();
-names[4] = names[4].substring();
+names[0] = names[0].substring(0,names[0].split(" ")[0].length);
+names[1] = names[1].substring(0,names[1].split(" ")[0].length);
+names[2] = names[2].substring(0,names[2].split(" ")[0].length);
+names[3] = names[3].substring(0,names[3].split(" ")[0].length);
+names[4] = names[4].substring(0,names[4].split(" ")[0].length);
```
Comment (Buroni) [inline 1-exercises/J-string-substring/exercise2.js:21]: The expression `names[index].split(" ")[0].length` is repeated 5 times here -- can you think of a way to put it in a re-usable function?

File: 2-mandatory/1-oxygen-levels.js [hunk 3]
```diff
@@ -11,7 +11,11 @@
     Some string methods that might help you here are .replace() and .substring(). 
 */
 
-function safeLevels() {}
+function safeLevels(oxygenLevels) {
+  let oxygenLevel =oxygenLevels.map(level => parseFloat(level.substring(0, level.length - 1)));
```
Comment (Buroni) [inline 2-mandatory/1-oxygen-levels.js:15]: Nice!

One note for improvement -- `oxygenLevel` isn't a clear variable name. A better name could be `oxygenLevelNumbers`, as you are mapping the percentage strings onto an array of numbers.

File: 2-mandatory/3-space-colonies.js [hunk 4]
```diff
@@ -15,7 +15,13 @@
   
 */
 
-function colonisers() {}
+function colonisers(voyager) {
+  let voyFamily = voyager.filter(
+    (voy) =>
+      voy.substring(voy.length - 6) === "family" && voy.substring(0, 1) === "A"
+  );
```
Comment (Buroni) [inline 2-mandatory/3-space-colonies.js:22]: Very clear and good use of `substring`

File: 2-mandatory/7-password-validator.js [hunk 5]
```diff
@@ -25,51 +25,59 @@ PasswordValidationResult=  [false, false, false, false, true]
...
 
-const passwords1 = ["Se%5", "TktE.TJTU", "384#HsHF", "dvyyeyy!5", "tryT3729"]
-const passwords2 = ["StUFf27%", "Pl3nty!", "Jai33", "shajsaUA**&&", "Pl3nty!"]
+const passwords1 = ["Se%5", "TktE.TJTU", "384#HsHF", "dvyyeyy!5", "tryT3729"];
+const passwords2 = ["StUFf27%", "Pl3nty!", "Jai33", "shajsaUA**&&", "Pl3nty!"];
 
-const util = require('util');
+const util = require("util");
 
 function test(test_name, actual, expected) {
```
Comment (Buroni) [inline 2-mandatory/7-password-validator.js:63]: Your final test case fails here

**Review by Buroni — COMMENTED** [review]
Great job Somayeh! Clean code, good understanding of array methods, and all of your tests pass except one. See below for feedback comments.

**Pull request closed without merging**